src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -275,13 +275,13 @@ private static JObject ReadProject(string projectJsonPath)
            } 
         }
 
-        private JToken GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
+        private JObject GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
         {
             if(string.IsNullOrWhiteSpace(framework))
             {
-                return projectJsonRoot["dependencies"];
+                return (JObject) projectJsonRoot["dependencies"];
             }
-            return projectJsonRoot.SelectToken("frameworks." + NewtonsoftEscapeJProperty(framework) + ".dependencies");
+            return (JObject) projectJsonRoot["frameworks"][framework]["dependencies"];
```

Comment:
The other two occurrences look like they want to determine the path to the entry, and so use the NewtonsoftEscapeJProperty method to construct it and refer to it as a JProperty. Here I wanted to retrieve the JObject that is referenced at this path.